mmv1/products/vertexai/FeatureOnlineStoreFeatureview.yaml

```diff
@@ -118,3 +124,62 @@ properties:
         description: |
           Columns to construct entityId / row keys. Start by supporting 1 only.
         item_type: Api::Type::String
+  - !ruby/object:Api::Type::NestedObject
+    name: 'vectorSearchConfig'
+    description: |
+      Configuration for vector search. It contains the required configurations to create an index from source data, so that approximate nearest neighbor (a.k.a ANN) algorithms search can be performed during online serving.
+    immutable: true
+    min_version: beta
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'embeddingColumn'
+        required: true
+        description: |
+          Column of embedding. This column contains the source data to create index for vector search. embeddingColumn must be set when using vector search.
+      - !ruby/object:Api::Type::Array
+        name: 'filterColumns'
+        description: |
+          Columns of features that're used to filter vector search results.
+        item_type: Api::Type::String
+      - !ruby/object:Api::Type::String
+        name: 'crowdingColumn'
+        description: |
+          Column of crowding. This column contains crowding attribute which is a constraint on a neighbor list produced by nearest neighbor search requiring that no more than some value k' of the k neighbors returned have the same value of crowdingAttribute.
+      - !ruby/object:Api::Type::Enum
+        name: distanceMeasureType
+        description: |
+          The distance measure used in nearest neighbor search.
+          SQUARED_L2_DISTANCE: Euclidean (L_2) Distance.
+
+          COSINE_DISTANCE: Cosine Distance. Defined as 1 - cosine similarity.
+
+          DOT_PRODUCT_DISTANCE: Dot Product Distance. Defined as a negative of the dot product.
```

Comment:
Do you also want to include "We strongly suggest using DOT_PRODUCT_DISTANCE + UNIT_L2_NORM instead of COSINE distance. Our algorithms have been more optimized for DOT_PRODUCT distance which, when combined with UNIT_L2_NORM, is mathematically equivalent to COSINE distance and results in the same ranking."

Alternatively, you could link to the [enum definition](https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.featureOnlineStores.featureViews#DistanceMeasureType) for more details. Something like:

```suggestion
          For details on allowed values, see the [API documentation](https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.featureOnlineStores.featureViews#DistanceMeasureType).
```

Reply:
Note that as written, this will display in the docs as:

![Screenshot 2024-01-17 at 10 44 47 AM](https://github.com/GoogleCloudPlatform/magic-modules/assets/299979/2a7f4317-29a7-4ae0-9c86-5bba488ce6d6)

https://googlecloudplatform.github.io/magic-modules/develop/resource/#add-documentation has details on previewing documentation.

Reply:
Right. I've just linked to the API documentation as suggested.